### PR TITLE
Changing context in monitoring dashboard

### PR DESCRIPTION
**Pull request by sanketpathak**
Gherkin Script for epic ODC-3898
Story: https://issues.redhat.com/browse/ODC-4226

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+   And user selects one workload at a time
+   And check for charts of the workload
+   And user refresh the page
+   Then the page should auto select the previously selected workload
+
+@regression
+Scenario: Checking filter dropdown for workloads on Monitoring Dashboard from workloads
+   Given user is in developer perspective
+   When user clicks on dropdown which would allow user to filter out workloads
+   And user selects All Workloads at a time
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:21]:
```suggestion
   And user selects All Workloads
```

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+   And user selects one workload at a time
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:22]:
```suggestion
   And user selects one workload
```

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+Background:
+    Given user logged into the openshift application
+
+@regression
+Scenario: Checking filter dropdown for workloads on Monitoring Dashboard coming from workloads
+   Given user is in developer perspective
+   And dropdown by default have All Workloads selected
+   When user clicks on dropdown which would allow user to filter out workloads
+   And user selects one workload at a time
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:12]:
```suggestion
   And user selects one workload
```

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+   And user refresh the page
+   Then the page should auto select the previously selected workload
+
+@regression
+Scenario: Checking filter dropdown for workloads on Monitoring Dashboard from workloads
+   Given user is in developer perspective
+   When user clicks on dropdown which would allow user to filter out workloads
+   And user selects All Workloads at a time
+   And user selects one workload at a time
+   And check for charts of the workload/workloads
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:23]:
```suggestion
   And user checks for charts of the workload/workloads
```

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+   And user switches to a different project from project selector 
+   Then filter dropdown option should be set to All workloads
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:25]: Please do not use should. Rearrange the then statement

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+	User should be able I want to change the filter in the Monitoring Dashboard
+
+Background:
+    Given user logged into the openshift application
+    And user is at the developer perspective
+
+@regression
+Scenario: Checking filter dropdown for workloads on Monitoring Dashboard navigating from workloads
+   Given user is in topology view with workloads
+   And user clicks monitoring dashboard from the sidebar of workload
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:11]: Given statement should not perform any action

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+   And user checks filter dropdown by default have "All Workloads" selected
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:12]: Given statement should not perform any action

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 (hunk 6 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:11]: It's not performing action just having a given condition

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 (hunk 7 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:12]: Same as above

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 8]
```diff
@@ -0,0 +1,29 @@
+Feature: Changing context in Monitoring Dashboard
+	User should be able I want to change the filter in the Monitoring Dashboard
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:2]: Please change the sentence as "User should be able to change the filter in the Monitoring Dashboard"

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+Background:
+    Given user logged into the openshift application
+    And user is at the developer perspective
+
+@regression
+Scenario: Checking filter dropdown for workloads on Monitoring Dashboard navigating from workloads
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:9]: I think scenario description and step looks different

In Scenario description, You mentioned "filter dropdown values verification for workloads in dashboard
But in steps, when statement says refreshes the page..  
and Then statement says workloads auto selection

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+   And user checks for charts of the workload
+   And user refreshes the page
+   Then the page will auto select the previously selected workload
+
+@regression
+Scenario: Checking filter dropdown for workloads on Monitoring Dashboard navigating from nav bar
+   Given User is at the Monitoring dashboard
+   When user clicks on filter dropdown
+   And user selects "All Workloads"
+   And user can check charts for all workloads
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:24]: This is something an outcome I believe.. Can you split the scenarios as required

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+Background:
+    Given user logged into the openshift application
+    And user is at the developer perspective
+
+@regression
+Scenario: Checking filter dropdown for workloads on Monitoring Dashboard navigating from workloads
+   Given user is in topology view with workloads
+   When user clicks monitoring dashboard from the sidebar of workload
+   And user checks filter dropdown by default have "All Workloads" selected
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:12]: If user comes from topology sidebar link, then filter dropdown should be autopopulated with the selected workload in monitoring page

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+   And user clicks on dropdown
+   And user selects one workload
+   And user checks for charts of the workload
+   And user refreshes the page
+   Then the page will auto select the previously selected workload
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:17]: Can we split this into a new scenario, where user manually selects the workload using filter dropdown?

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+@regression
+Scenario: Checking filter dropdown for workloads on Monitoring Dashboard navigating from nav bar
+   Given User is at the Monitoring dashboard
+   When user clicks on filter dropdown
+   And user selects "All Workloads"
+   And user can check charts for all workloads
+   And user selects one workload
+   And user checks for charts of the workload/workloads
+   And user switches to a different project from project selector 
+   Then filter dropdown option will be set to "All workloads"
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:28]: Add Proper indendation to the keywords Given, When & And steps.

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 (hunk 9 above)
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:9]:
```suggestion
Scenario: Auto selection of filter dropdown when navigating from topology 
```

Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:9]: It's just not checking for auto-selection only hence the title

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 (hunk 12 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:17]: We can but redundancy will increase

File: frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898 [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    And user refreshes the page
+    Then the page will auto select the previously selected workload
+
+@regression
+Scenario: Checking filter dropdown for workloads on Monitoring Dashboard on project change
+    Given user is at the Monitoring dashboard
+    When user clicks on filter dropdown
+    And user selects one workload from the list
+    And user switches to a different project from project selector 
+    And filter dropdown option will be set to "All workloads"
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/monitoring/Changing_context_in_Monitoring_Dashboard_3898:48]: Update it as Then statement

**Pull request merged**